src/features/generic_events/aggregate/user-actions/user-actions-aggregator.js

```diff
@@ -0,0 +1,95 @@
+import { generateSelectorPath } from '../../../../common/dom/selector-path'
+import { OBSERVED_WINDOW_EVENTS } from '../../constants'
+import { AggregatedUserAction } from './aggregated-user-action'
+
+export class UserActionsAggregator {
+  /** @type {AggregatedUserAction=} */
+  #aggregationEvent = undefined
+  #aggregationKey = ''
+  #subscribers = []
+
+  /**
+   * @param {Function} storeMethod The function given by the parent class to call when an event is done aggregating
+   */
+  constructor () {
+    return this
+  }
+
+  #emit (event, data) {
+    this.#subscribers.forEach(subscriber => {
+      if (subscriber.event === event) {
+        subscriber.callback(data)
+      }
+    })
+  }
+
+  on (event, callback) {
+    this.#subscribers.push({ event, callback })
+  }
+
+  /**
+   * Process the event and determine if a new aggregation set should be made or if it should increment the current aggregation
+   * @param {Event} evt The event supplied by the addEventListener callback
+   */
+  process (evt) {
+    const selectorPath = getSelectorPath(evt)
+    const aggregationKey = getAggregationKey(evt, selectorPath)
+    if (aggregationKey === this.#aggregationKey) {
+      // an aggregation exists already, so lets just continue to increment
+      this.#aggregationEvent.aggregate(evt)
+    } else {
+      // store the prev existing one (if there is one)
+      this.storeCurrentUserActionInFeature()
+      // then set as this new event aggregation
+      this.#aggregationKey = aggregationKey
+      this.#aggregationEvent = new AggregatedUserAction(evt, selectorPath)
+    }
+  }
+
+  /**
+   * Store the current aggregation set in the parent feature's event buffer and clear the state of the aggregator
+   */
+  storeCurrentUserActionInFeature () {
+    // store the prev existing one (if there is one)
+    if (this.#aggregationEvent) {
+      this.#emit('aggregation-complete', this.#aggregationEvent)
+      // then clear it...
+      this.#aggregationKey = ''
+      this.#aggregationEvent = undefined
+    }
+  }
+}
+
+/**
+ * Generates a selector path for the event, starting with simple cases like window or document and getting more complex for dom-tree traversals as needed.
+ * Will return a random selector path value if no other path can be determined, to force the aggregator to skip aggregation for this event.
+ * @param {Event} evt
+ * @returns {string}
+ */
+function getSelectorPath (evt) {
+  let selectorPath
+  if (OBSERVED_WINDOW_EVENTS.includes(evt.type) || evt.target === evt.target?.top) selectorPath = 'window'
```

Comment:
Split out a case for `iframe window` as target.
```suggestion
  if (evt.target === window) ... 
```

Reply:
Actually just make an `isIframe` etc property on user actions event if it happened in an iframe.